src/Avalonia.Controls.TreeDataGrid/TreeDataGrid.cs

```diff
@@ -239,6 +240,16 @@ protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
             Scroll = e.NameScope.Find<ScrollViewer>("PART_ScrollViewer");
         }
 
+        protected void OnPreviewKeyDown(object? o, KeyEventArgs e)
+        {
+            if (e.Key == Key.PageDown || e.Key == Key.PageUp)
+            {
+                e.Handled = true;
```

Comment:
I wonder, won't it break some apps, if tree data grid will intercept these keydown events even when it can't actually handle it? Let's imagine, if last item is already selected, and user presses PageDown to do some another operation in the app (scroll global scrollviewer or anything else).

Isn't it better to return bool from _selection?.OnPreviewKeyDown and decide if it's handled or not? Or set handled inside of OnPreviewKeyDown.

Reply:
I don't think it may be needed. The only additional case it may be not handled here when childrenCount is 0,otherwise it would be handled anyways. Check for childrencount I can add here I think